File: src/main/java/kr/co/lunatalk/domain/productlike/domain/ProductLike.java
package kr.co.lunatalk.domain.productlike.domain;

import jakarta.persistence.*;
import kr.co.lunatalk.domain.common.domain.BaseTimeEntity;
import kr.co.lunatalk.domain.member.domain.Member;
import kr.co.lunatalk.domain.product.domain.Product;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "product_likes", uniqueConstraints = {
	@UniqueConstraint(columnNames = {"member_id", "product_id"})
})
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ProductLike extends BaseTimeEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "member_id", nullable = false)
	private Member member;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "product_id", nullable = false)
	private Product product;

	@Builder(access = AccessLevel.PRIVATE)
	private ProductLike(Member member, Product product) {
		this.member = member;
		this.product = product;
	}

	public static ProductLike create(Member member, Product product) {
		return ProductLike.builder()
			.member(member)
			.product(product)
			.build();
	}
}

File: src/main/kotlin/kr/co/lunatalk/domain/productlike/domain/ProductLike.kt
package kr.co.lunatalk.domain.productlike.domain

import jakarta.persistence.*
import kr.co.lunatalk.domain.common.domain.BaseTimeEntity
import kr.co.lunatalk.domain.member.domain.Member
import kr.co.lunatalk.domain.product.domain.Product

@Entity
@Table(
    name = "product_likes",
    uniqueConstraints = [UniqueConstraint(columnNames = ["member_id", "product_id"])]
)
open class ProductLike protected constructor() : BaseTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    open var member: Member? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    open var product: Product? = null
        protected set

    private constructor(member: Member, product: Product) : this() {
        this.member = member
        this.product = product
    }

    companion object {
        fun create(member: Member, product: Product): ProductLike {
            return ProductLike(member, product)
        }
    }
}
